common/lib/xmodule/xmodule/modulestore/mixed.py

```diff
@@ -172,8 +172,12 @@ def _get_modulestore_for_courseid(self, course_id=None):
             if mapping is not None:
                 return mapping
             else:
+                if isinstance(course_id, LibraryLocator):
```

Comment:
This method, `_get_modulestore_for_courseid` is key to how MixedModulestore works and is used by almost every other method in this file, so it would create a ton of code redundancy to move this branching logic anywhere else I think. It would be more clear if it were called `_get_modulestore_for_course_or_library` but that's too long and this is one of just many cases where the code that uses it (methods like `get_item`, `update_item` etc.) doesn't care if they're dealing with courses or libraries and shouldn't have to know about the difference.